proxy/api/src/ethereum_address.rs

```diff
@@ -0,0 +1,35 @@
+use serde::{Deserialize, Serialize};
+use std::convert::TryFrom;
+
+/// An [EIP-55](https://eips.ethereum.org/EIPS/eip-55) Ethereum address.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[serde(try_from = "String", into = "String")]
+pub struct EthereumAddress(String);
+
+#[derive(Clone, Debug, thiserror::Error)]
+#[error("Invalid EIP-55 address")]
```

Comment:
Someone forgot their own proposal for lowercase errors ;)

Reply:
You got me, fixed :laughing: Some errors in `proxy/api/src/error.rs` are uppercase too, should I fix them now or in a separate PR to avoid noise?

Reply:
Ya, let's not add noise but thanks for calling that out!